### PR TITLE
[do not merge] COUNT API proposal

### PR DESCRIPTION
**Pull request by dconeybe**

### CODE REVIEW

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@class FIRQuery;
+
+NS_ASSUME_NONNULL_BEGIN
+
+NS_SWIFT_NAME(AggregateQuery)
+@interface FIRAggregateQuery : NSObject
+/** :nodoc: */
+- (id)init __attribute__((unavailable("FIRAggregateQuery cannot be created directly.")));
+
+@property(nonatomic, strong, readonly) FIRQuery *query;
```
Comment (ncooke3) [inline Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h:31]: `strong` can be removed. It is typically not featured in the signature.
```suggestion
@property(nonatomic, readonly) FIRQuery *query;
```

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+#pragma mark - Retrieving Data
+
+- (void)aggregationWithSource:(FIRAggregateSource)source completion:(void (^)(FIRAggregateQuerySnapshot *_Nullable snapshot, NSError *_Nullable error))completion NS_SWIFT_NAME(get(source:completion:));
```
Comment (ncooke3) [inline Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h:35]: Updated the Swift name...
```suggestion
- (void)aggregationWithSource:(FIRAggregateSource)source 
                                  completion:(void (^)(FIRAggregateQuerySnapshot *_Nullable snapshot, NSError *_Nullable error))completion 
                                  NS_SWIFT_NAME(aggregation(source:completion:));
```

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+@class FIRAggregateQuerySnapshot;
+@class FIRQuery;
+
+NS_ASSUME_NONNULL_BEGIN
+
+NS_SWIFT_NAME(AggregateQuery)
+@interface FIRAggregateQuery : NSObject
+/** :nodoc: */
+- (id)init __attribute__((unavailable("FIRAggregateQuery cannot be created directly.")));
```
Comment (ncooke3) [inline Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuery.h:29]: I often see this pattern like:
```suggestion
- (id)init NS_UNAVAILABLE;
```

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#import <Foundation/Foundation.h>
+
+@class FIRAggregateQuery;
+
+NS_ASSUME_NONNULL_BEGIN
+
+NS_SWIFT_NAME(AggregateQuerySnapshot)
+@interface FIRAggregateQuerySnapshot : NSObject
+/** :nodoc: */
+- (id)init __attribute__((unavailable("FIRAggregateQuerySnapshot cannot be created directly.")));
```
Comment (ncooke3) [inline Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h:26]:
```suggestion
- (id)init NS_UNAVAILABLE;
```

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+@property(nonatomic, strong, readonly) FIRAggregateQuery *query;
+
+- (nullable NSNumber*)count;
```
Comment (ncooke3) [inline Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h:30]: I'm not sure if this is a good idea or not, but have you considered always returning count for a given aggregate query?

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+@class FIRAggregateQuery;
+
+NS_ASSUME_NONNULL_BEGIN
+
+NS_SWIFT_NAME(AggregateQuerySnapshot)
+@interface FIRAggregateQuerySnapshot : NSObject
+/** :nodoc: */
+- (id)init __attribute__((unavailable("FIRAggregateQuerySnapshot cannot be created directly.")));
+
+@property(nonatomic, strong, readonly) FIRAggregateQuery *query;
```
Comment (ncooke3) [inline Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h:28]: `strong` can be removed.
```suggestion
@property(nonatomic, readonly) FIRAggregateQuery *query;
```

File: Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h (hunk 5 above)
Comment (dconeybe) [reply Firestore/Source/Public/FirebaseFirestore/FIRAggregateQuerySnapshot.h:30]: Yes, we have considered always returning count for a given aggregate query. It is _not_ something that we will support. We decided that the client must explicitly request the count. So in a future world where customers can request various aggregations, if they don't request count then `count` will return `nil`.

**Comment by paulb777** [comment]
Closing. Please reopen if needed.

**Pull request closed without merging**